packages/astro/src/@types/astro.ts

```diff
@@ -1601,25 +1601,29 @@ export interface AstroUserConfig {
 	experimental?: {
 		/**
 		 * @docs
-		 * @name experimental.optimizeHoistedScript
+		 * @name experimental.directRenderScript
 		 * @type {boolean}
 		 * @default `false`
-		 * @version 2.10.4
+		 * @version 4.5.0
 		 * @description
-		 * Prevents unused components' scripts from being included in a page unexpectedly.
-		 * The optimization is best-effort and may inversely miss including the used scripts. Make sure to double-check your built pages
-		 * before publishing.
-		 * Enable hoisted script analysis optimization by adding the experimental flag:
+		 * Directly render scripts as declared in Astro files (with features like TypeScript, importing `node_modules`, and deduplicating
```

Comment:
Will review this after the changeset is final, just in case any wording used there makes you want to update something here first!